exchanges/binance/binance_websocket.go

```diff
@@ -95,18 +84,20 @@ func (b *Binance) WsConnect() error {
 	}
 
 	go b.wsReadData()
-	go b.KeepAuthKeyAlive()
-	return nil
+
+	subs, err := b.GenerateSubscriptions()
+	if err != nil {
+		return err
+	}
+	return b.Websocket.SubscribeToChannels(subs)
 }
 
 // KeepAuthKeyAlive will continuously send messages to
 // keep the WS auth key active
 func (b *Binance) KeepAuthKeyAlive() {
 	b.Websocket.Wg.Add(1)
-	defer func() {
-		b.Websocket.Wg.Done()
-	}()
-	ticks := time.NewTicker(time.Minute * 30)
+	defer b.Websocket.Wg.Done()
+	ticks := time.NewTicker(time.Hour)
```

Comment:
Okay, there's a difference with their updated perps docs:
https://binance-docs.github.io/apidocs/futures/en/#user-data-streams
```
Keepalive a user data stream to prevent a time out. User data streams will close after 60 minutes. It's recommended to send a ping about every 60 minutes.
```

Reply:
I'm still not going to say that you should leave it at the max allowed level which just leaves us too open to miss the allowance. Please change to less than 60